### PR TITLE
Adds Elasticache smart client and switches from Redis to it

### PR DESCRIPTION
**Pull request by elgris**
More or less resembles Redis store.
I did not use exact Memcached client, but insted used "smarter" version of it tailored to AWS Elasticache. Should be easier to use with AWS Elasticache cluster

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -208,6 +208,11 @@
             <artifactId>jedis</artifactId>
             <version>2.9.0</version>
         </dependency>
+        <dependency>
+            <groupId>com.googlecode.xmemcached</groupId>
```
Comment (lmineiro) [inline pom.xml:212]: Is this a dep for the AWS 'smart" client?

Comment (elgris) [reply pom.xml:212]: right: https://github.com/killme2008/xmemcached/wiki/User-Guide#aws-elasticcache-client

File: src/main/java/org/kairosdb/datastore/cassandra/cache/persistence/ElastiCacheConfiguration.java [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+    @Named(WRITER_THREADS)
+    private int writerThreads = 20;
+
+    @Inject(optional = true)
+    @Named(WRITER_THREAD_IDLE_TIMEOUT_SECONDS)
+    private int workerThreadIdleTimeoutSeconds = 300;
+
+    @Inject(optional = true)
+    @Named(CONNECTION_POOL_SIZE)
+    private int connectionPoolSize = 50;
```
Comment (lmineiro) [inline src/main/java/org/kairosdb/datastore/cassandra/cache/persistence/ElastiCacheConfiguration.java]: If the writer threads are only 20 I don't think this needs to be 50. I'd change this to default to the same size as the writer threads

**Comment by lmineiro** [comment]
👍

**Comment by lmineiro** [comment]
shipit

**Pull request merged**